### ISSUE TITLE
Getting a queryset of all pages/snippets for the current locale

### ISSUE THREAD

**Issue by TheBoroer** [wagtail/wagtail-localize#421]
Hey,
Sorry if I've missed this somewhere in the docs or examples but I've spent 2 days trying to figure this out now and i'm kinda lost 😅 

What I'm trying to do is access all the pages of a specific type for the current/active locale to build out a context variable to then pass into the template for rendering.

Using something like this `Lesson.objects.all()` now gives me all the Lesson pages in across every locale.

Is there a wagtail/wagtail-localize way or method I can get a queryset of all the pages/snippets/items in the current locale and if there's items that haven't been translated yet (but the locale is synced with english) to have it also include those english items as well?

Or is my only option to make customer helper functions to do this for me?

**Comment by saevarom** [comment]
What was the answer to your question @TheBoroer ?

**Comment by codesentry17** [comment]
I need the answer my friend, I'm stuck on the very same issue.

**Comment by TheBoroer** [comment]
Hey! Whoa i didn't see the earlier replies to this... 
I actually don't remember what I ended up doing for this 😅 ... but I believe I did this for pages:
`Page.objects.objects.filter(locale=locale_goes_here).specific()`
and for snippets I had to make a helper function that would do a similar query as above (minus the .specific())  for each snippet model and then i'd convert the Queryset into a list and concactenate/union the lists so that all snippets were in 1 list.

I hope that helps!
